src/Alert/Alert.tsx

```diff
@@ -0,0 +1,194 @@
+/** @jsx jsx */
+import { jsx, ClassNames } from "@emotion/core";
+import React, { CSSProperties, Fragment } from "react";
+import PropTypes from "prop-types";
+import classnames from "classnames";
+
+import { base } from "../typography";
+import { IconClose } from "../icons/IconClose";
+import { colors, PaletteColor } from "../colors";
+import { getOffsetInPalette } from "../colors/utils/getOffsetInPalette";
+
+interface AlertProps {
+  /**
+   * color theme for alert
+   * @default "light"
+   */
+  theme?: "light" | "dark";
+
+  /**
+   * The color used for the heading icon, the heading text will be 2 shades
+   * darker if light theme and 2 shaders lighter if a dark theme
+   */
+  color: PaletteColor;
+
+  /**
+   * The icon displayed to the left of the heading text
+   */
+  Icon: React.ComponentType<{ className?: string }>;
+
+  heading: React.ReactNode;
+
+  /**
+   * actions could be a button
+   * or a tooltip or anything the Alert should display after the children
+   */
+  actions?: React.ReactNode;
+
+  /**
+   * The content of the card, appears below the title
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Override how the `header` is rendered. You can pass either an intrinisic
+   * jsx element as a string (like "h1") or a react element (`<h1 />`)
+   *
+   * If you pass a react element, props that we add are spread onto the input.
+   *
+   * @default "h2"
+   */
+  headingAs?: React.ReactElement | keyof JSX.IntrinsicElements;
+
+  /**
+   * callback for handling the clicks on the close button.
+   */
+  onClose: () => void;
+
+  /**
+   * layout for longer content
+   *
+   * @default false
+   */
+  extended?: boolean;
+
+  className?: string;
+  style?: CSSProperties;
+}
+
+export const Alert: React.FC<AlertProps> = ({
+  heading,
+  onClose,
+  actions,
+  headingAs = "h2",
+  children,
+  color,
+  Icon,
+  theme = "light",
+  extended = false,
+  ...otherProps
+}) => {
+  return (
+    <section
+      {...otherProps}
+      css={{
+        backgroundColor:
+          theme === "light" ? colors.white : colors.black.lighter,
+        color: theme === "light" ? colors.black.base : colors.white,
```

Comment:
This doesn't take into account that we might have other themes that aren't "dark" or "light", but that's not very likely so that's fine. If we wanted to protect ourselves from this then I'd make the following change:

<details>
<summary>expand diff</summary>

```diff
diff --git a/src/Alert/Alert.tsx b/src/Alert/Alert.tsx
index d22dd52..ec8e791 100644
--- a/src/Alert/Alert.tsx
+++ b/src/Alert/Alert.tsx
@@ -66,6 +66,11 @@ interface AlertProps {
   style?: CSSProperties;
 }
 
+// @see https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking
+function assertUnreachable(x: never): never {
+  throw new TypeError(`Didn't expect to get here ${x}`);
+}
+
 export const Alert: React.FC<AlertProps> = ({
   heading,
   onClose,
@@ -83,8 +88,17 @@ export const Alert: React.FC<AlertProps> = ({
       {...otherProps}
       css={{
         backgroundColor:
-          theme === "light" ? colors.white : colors.black.lighter,
-        color: theme === "light" ? colors.black.base : colors.white,
+          theme === "light"
+            ? colors.white
+            : theme === "dark"
+            ? colors.black.lighter
+            : assertUnreachable(theme),
+        color:
+          theme === "light"
+            ? colors.black.base
+            : theme === "dark"
+            ? colors.white
+            : assertUnreachable(theme),
         boxShadow: `0 4px 8px 0 rgba(0, 0, 0, .04)`,
         borderStyle: "solid",
         borderRadius: 4,
```
</details>

Then if we add another option to `theme`, TypeScript will recognize we need to add cases here.

You don't have to make a change, just pointing thi sout.

Reply:
I'm definitely for this. 
There were a couple assertUnreachable in the codebase already so I made a common one in shared. If there was a reason they were in each file I can definitely split them up again 👍